app/assets/javascripts/discourse/tests/integration/components/pinned-options-test.gjs

```diff
@@ -61,11 +62,15 @@ module("Integration | Component | select-kit/pinned-options", function (hooks) {
       </template>
     );
 
-    assert.strictEqual(this.subject.header().name(), "unpinned");
+    assert
+      .dom(".pinned-options-trigger-btn .d-button-label")
+      .hasText("Unpinned");
+
+    await click(".pinned-options-trigger-btn");
 
-    await this.subject.expand();
-    await this.subject.selectRowByValue("pinned");
+    await click('[data-pinned-state="pinned"]');
 
-    assert.strictEqual(this.subject.header().name(), "pinned");
+    assert.true(this.topic.pinned, "topic should be pinned");
+    assert.false(this.topic.unpinned, "topic should not be marked as unpinned");
```

Comment:
a very minor nit - in specs/tests it's preferred to avoid "should" and just declare the desired state:

```suggestion
    assert.true(this.topic.pinned, "topic is pinned");
    assert.false(this.topic.unpinned, "topic isn't unpinned");
```